staging/src/k8s.io/api/auditregistration/v1alpha1/types.go

```diff
@@ -90,41 +90,68 @@ type AuditSinkList struct {
 	// +optional
 	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
 
-	// List of audit configurations.
+	// `items` is a list of `auditSink` configurations.
 	Items []AuditSink `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
 
 // Policy defines the configuration of how audit events are logged
 type Policy struct {
-	// The Level that all requests are recorded at.
+	// `level` is the base level that all requests are recorded at.
 	// available options: None, Metadata, Request, RequestResponse
 	// required
 	Level Level `json:"level" protobuf:"bytes,1,opt,name=level"`
 
-	// Stages is a list of stages for which events are created.
+	// `stages` is a list of stages for which events are created.
+	// If no stages are given nothing will be logged
 	// +optional
 	Stages []Stage `json:"stages" protobuf:"bytes,2,opt,name=stages"`
+
+	// `rules` define how `auditClass` objects should be handled.
+	// A request may fall under multiple `auditClass` objects.
+	// Rules are evaluated in order (first matching wins).
+	// Rules take precedence over the the parent `level` & `stages`.
+	// Unmatched requests use the parent `level` rule.
+	// +optional
+	Rules []PolicyRule `json:"rules" protobuf:"bytes,3,opt,name=rules"`
+}
+
+// PolicyRule defines what level and stages are recorded for an `auditClass`.
+type PolicyRule struct {
+	// `withAuditClass` is the name of the `auditClass` object. This rule matches requests that are
+	// classified with this `auditClass`
+	WithAuditClass string `json:"withAuditClass" protobuf:"bytes,1,opt,name=withAuditClass"`
+
+	// `level` that all requests this rule applies to are recorded at.
+	// available options: None, Metadata, Request, RequestResponse
+	// This will override the parent policy Level.
+	// required
+	Level Level `json:"level" protobuf:"bytes,2,opt,name=level"`
+
+	// `stages` is a list of stages for which events are created.
+	// If no stages are given nothing will be logged.
```

Comment:
Does this works as a union with the Policy stages as in the audit API or it completely overrides it? This should also be made explicit.